### PR TITLE
Exposed More Info to gameInfo

### PR DESCRIPTION
**Pull request by devingearing**
By exposing more info we can allow for third party extension to read this info to make overlays or whatever they want without needing them to read localStorage.

### CODE REVIEW

File: src/battle-scene.ts [hunk 1]
```diff
@@ -1993,11 +1998,15 @@ export default class BattleScene extends SceneBase {
...
+			currentPhase: (this.getCurrentPhase())?.constructor.name || "Unknown",
 			biome: this.currentBattle ? getBiomeName(this.arena.biomeType) : '',
 			wave: this.currentBattle?.waveIndex || 0,
-			party: this.party ? this.party.map(p => {
-				return { name: p.name, level: p.level };
-			}) : []
+			 party: this.party ? this.party.map(p => {
+				return { name: p.name, level: p.level, speciesId: p.species.speciesId, fusionId: p.fusionSpecies?.speciesId || -1, ability: p.abilityIndex, nature:p.nature, ivs: p.ivs, active: p.active, gender: p.gender, pokerus: p.pokerus,fieldPosition: p.fieldPosition};
+			}) : [],
+			enemyParty: this.getEnemyParty() ? this.getEnemyParty().map(p => {
```
Comment (XGSleepWalker) [inline src/battle-scene.ts:2007]: https://github.com/pagefaultgames/pokerogue/pull/647/files#diff-4b1a5f9160d1d8aecce0db85774f20e3c254f97c27cc0d0473911d5cbd565396 - Check this as a reference, but the main point is that gameInfo shouldn't expose more info than what the player can see. Exposing enemy IVs goes against the purpose of IV Scanner, and likewise for ability.

Comment (devingearing) [reply src/battle-scene.ts:2007]: That is a valid point. If I removed the ivs and nature would this look alright then?

Comment (devingearing) [reply src/battle-scene.ts]: Here is video proof


Uploading 2024-05-16 20-17-51 (1).mp4…

**Review by Admiral-Billy — APPROVED** [review]
I tested locally and everything seems to work as you'd expect (at least from an output standpoint).

Here's a summary on the discord: https://discord.com/channels/1125469663833370665/1240785598214705192/1240863471483097149

File: src/battle-scene.ts [hunk 2]
```diff
@@ -1533,7 +1533,15 @@ export default class BattleScene extends SceneBase {
...
-
+		if(this.getCurrentPhase()?.constructor.name === 'SwitchSummonPhase'){
+			this.updateGameInfo();
+		}
+		if(this.getCurrentPhase()?.constructor.name === 'SummonPhase'){
+			this.updateGameInfo();
+		}
+		if(this.getCurrentPhase()?.constructor.name === 'TurnInitPhase'){
+			this.updateGameInfo();
+		}
```
Comment (Xavion3) [inline src/battle-scene.ts:1544]: This could just be an includes, makes it easier to add more later.

**Comment by flx-sta** [comment]
No action within more than a month. 
Considered stale.

Feel free to re-open it when you are still working on it/want to work on it again

**Pull request closed without merging**